src/Sylius/Behat/Context/Api/Shop/ProductContext.php

```diff
@@ -44,6 +46,40 @@ public function iOpenProductPage(ProductInterface $product): void
         $this->client->show($product->getCode());
     }
 
+    /**
+     * @When /^I browse products from (taxon "([^"]+)")$/
+     */
+    public function iBrowseProductsFromTaxon(TaxonInterface $taxon): void
+    {
+        $this->client->index();
+        $this->client->addFilter('productTaxons.taxon.code', $taxon->getCode());
+        $this->client->filter();
+    }
+
+    /**
+     * @Then I should see the product price :productPrice
+     */
+    public function iShouldSeeTheProductPrice(string $productPrice): void
+    {
+        $price = (int) filter_var($productPrice, FILTER_SANITIZE_NUMBER_INT);
+
+        Assert::true($this->hasProductWithPrice([$this->responseChecker->getResponseContent($this->client->getLastResponse())], $price));
+    }
+
+    /**
+     * @Then I should see the product :product with price :productPrice
+     */
+    public function iShouldSeeTheProductWithPrice(ProductInterface $product, string $productPrice): void
+    {
+        $price = (int) filter_var($productPrice, FILTER_SANITIZE_NUMBER_INT);
+
+        Assert::true(
+            $this->hasProductWithPrice(
```

Comment:
imho it should be refactored a bit: i suggest to get product by `code` and compere its price with `$productPrice`, you will have only one `channelPricing` so you can get first element or get it by channel from sharedStorage

Reply:
getting channel from sharedStorage sounds great!